src/date/expand-pattern.js

```diff
@@ -33,10 +55,43 @@ return function( pattern, cldr ) {
 
 	switch ( true ) {
 		case "skeleton" in pattern:
-			result = cldr.main([
-				"dates/calendars/gregorian/dateTimeFormats/availableFormats",
-				pattern.skeleton
-			]);
+			var flagDate = false, flagTime = false;
+			var skeleton = pattern.skeleton;
+			for ( var i = 0 ; i < skeleton.length ; i++ ) {
+				var ch = skeleton[ i ];
+				if ( ch == 'h' || ch == 'H' || ch == 'm' || ch == 's' ) { //Check if skeleton contains Time component
+					flagTime = true;
+				}
+				if ( ch == 'y' || ch == 'M' || ch == 'd' ) { // Check if skeleton contains Date component
+					flagDate = true;
+				}
+			}
+			if ( ( flagDate != flagTime ) ) {
+				result = cldr.main([
+					"dates/calendars/gregorian/dateTimeFormats/availableFormats",
+					skeleton
+				]);
+			}
+			else if ( flagDate && flagTime ) {
+				if ( skeleton.indexOf( "MMMM" ) > -1 && skeleton.indexOf( "E" ) > -1 ) {
+					result = getDateTime( "full", cldr );
+				}
+				else if ( skeleton.indexOf( "MMMM" ) > -1 ) {
+					result = getDateTime( "long", cldr );
+				}
+				else if ( skeleton.indexOf( "MMM" ) > -1 ) {
+					result = getDateTime( "medium", cldr );
+				}
+				else {
+					result = getDateTime( "short", cldr );
+				}
+			}
+			else {
```

Comment:
If we check for all date and time fields in `flagDate` and `flagTime`, there should be no `else`, right?